### DISCUSSION TITLE
Filter Expressions

### DISCUSSION THREAD

**Discussion by Mdesmery** [CrowdStrike/falconpy#43] [Q&A] [answered]
I'm having trouble using the filter parameter.  Can you provide some sample filter expressions?

Specifically, I'd like to use the QueryDevicesByFilter command to find a specific host.  Among other things I've tried 

PARAMS = {
    'offset': 0,
    'limit': 100,
    'sort': 'hostname',
    'filter': 'hostname=TD1000623'
}
response = falcon.command('QueryDevicesByFilter', parameters=PARAMS)

which is a hostname in my environment, but I get an 'Invalid filter expression supplied' error.

**Comment by Mdesmery** [comment, marked as answer]
Ahhh, needs single quotes in the string.  This works!

    'filter': "hostname:'TD1000623'"

**Reply by jshcodes** [reply]
You found it before I managed to reply back. 😁
